R/taxize_nexml.R

```diff
@@ -6,38 +6,47 @@
 #' containing the corresponding identifier. 
 #' @param nexml a nexml object
 #' @param type the name of the identifier to use
-#' @param ... additional arguments (not implemented yet)
+#' @param ... additional arguments to `[taxald::get_ids()]`
 #' @export 
 #' @examples \dontrun{
 #' data(bird.orders)
 #' birds <- add_trees(bird.orders)
 #' birds <- taxize_nexml(birds, "NCBI")
 #' }
 # @importFrom taxize get_uid
-taxize_nexml <- function(nexml, type = c("NCBI"), ...){
+taxize_nexml <- function(nexml, type = c("ncbi", "itis", "col", "tpl",
+                                         "gbif", "wd"), ...){
   
   ## Soft dependency on taxize
-  if (!requireNamespace("taxize", quietly = TRUE)) {
-    stop("taxize package required to convert look up taxonomic ids",
+  if (!requireNamespace("taxald", quietly = TRUE)) {
+    stop("taxald package required to convert look up taxonomic ids",
          call. = FALSE)
   }
-  get_uid <- getExportedValue("taxize", "get_uid")
+  get_ids <- getExportedValue("taxald", "get_ids")
   
-  
-          type <- match.arg(type)
-          if(type == "NCBI"){
-            for(j in 1:length(nexml@otus)){
-              for(i in 1:length(nexml@otus[[j]]@otu)){
-                id <- get_uid(nexml@otus[[j]]@otu[[i]]@label)
-                if(is.na(id))
-                  warning(paste("ID for otu", nexml@otus[[j]]@otu[[i]]@label, "not found. Consider checking the spelling or alternate classification"))
-                else 
-                  nexml@otus[[j]]@otu[[i]]@meta <- New("ListOfmeta", list(
-                                 meta(href = paste0("http://ncbi.nlm.nih.gov/taxonomy/", id),
-                                      rel = "tc:toTaxon")))
+  type <- tolower(type)
+  type <- match.arg(type)
+
+  for(j in 1:length(nexml@otus)){
+    
+    # Resolve all ids at once
+    labels <- unname(vapply(nexml@otus[[j]]@otu, slot, character(1L), "label"))
+    taxa_ids <- get_ids(labels, type, format = "uri", ...)
+    
+    for(i in 1:length(taxa_ids)){
+      id <- taxa_ids[[i]]
+      if(is.na(id))
+        warning(paste("ID for otu", 
```

Comment:
Right, I just left this in for compatibility, but I agree this is a warning that has always been more annoying than helpful.  It's a lot less annoying than the previous one because taxald does a better job finding matches ... Ideally I think it might be preferable to collect these warnings in a log and return a single warning at the end?  Possibly also include a way to suppress the warning as a function argument.  I do feel it wouldn't be great to require inspecting the NeXML / meta elements manually to detect unresolved otu labels...